src/state/notes/reducer.js

```diff
@@ -51,8 +51,18 @@ export const noteLikes = (state = {}, { type, noteId, isLiked }) => {
     return state;
 };
 
-export const noteReads = (state = {}, { type, noteId }) =>
-    (types.SELECT_NOTE && noteId ? { ...state, [noteId]: true } : state);
+// Track notes that have been read and the filter they were read from
+export const noteReads = (state = {}, { type, noteId, filterName = 'all' }) => {
+    if (types.SELECT_NOTE === type) {
+        return noteId ? { ...state, [noteId]: filterName } : state;
+    }
+
+    if (types.RESET_READS === type) {
+        return mapValues(state, () => '');
+    }
+
+    return state;
+};
```

Comment:
Hmm ok. As far as I could tell this state was only keeping track of notes that were marked read locally, and was always set to `true`.

I believe in the previous version of the client we kept a separate array of notes that were read when on the unread filter, should we just do that again?